src/aiida_quantumespresso/workflows/xps.py

```diff
@@ -511,48 +531,59 @@ def prepare_structures(self):
+                    inputs[key] = node
+            if 'spglib_settings' in self.inputs:
+                spglib_settings = self.inputs.spglib_settings
+                inputs['spglib_settings'] = spglib_settings
+            else:
+                spglib_settings = None
+
+            result = get_xspectra_structures(input_structure, **inputs)
+
+            supercell = result.pop('supercell')
+            out_params = result.pop('output_parameters')
+            if out_params.get_dict().get('structure_is_standardized', None):
+                standardized = result.pop('standardized_structure')
+                self.out('standardized_structure', standardized)
+
+            # structures_to_process = {Key : Value for Key, Value in result.items()}
+            for site in ['output_parameters', 'supercell', 'standardized_structure']:
+                result.pop(site, None)
+            self.ctx.supercell = supercell
+            self.ctx.equivalent_sites_data = out_params['equivalent_sites_data']
+            self.out('supercell_structure', supercell)
+            self.out('symmetry_analysis_data', out_params)
+        elif self.ctx.atoms_list:
+            atoms_list = orm.List(self.ctx.atoms_list)
+            inputs = {
+                'atoms_list' : atoms_list,
+                'marker' : self.inputs.abs_atom_marker,
+                'metadata' : {
+                    'call_link_label' : 'get_marked_structures'
+                }
             }
-        } # populate this further once the schema for WorkChain options is figured out
-        if 'structure_preparation_settings' in self.inputs:
-            optional_cell_prep = self.inputs.structure_preparation_settings
-            for key, node in optional_cell_prep.items():
-                inputs[key] = node
-        if 'spglib_settings' in self.inputs:
-            spglib_settings = self.inputs.spglib_settings
-            inputs['spglib_settings'] = spglib_settings
-        else:
-            spglib_settings = None
-
-        if 'relax' in self.inputs:
-            relaxed_structure = self.ctx.relaxed_structure
-            result = get_xspectra_structures(relaxed_structure, **inputs)
-        else:
-            result = get_xspectra_structures(self.inputs.structure, **inputs)
-
-        supercell = result.pop('supercell')
-        out_params = result.pop('output_parameters')
-        if out_params.get_dict().get('structure_is_standardized', None):
-            standardized = result.pop('standardized_structure')
-            self.out('standardized_structure', standardized)
-
-        # structures_to_process = {Key : Value for Key, Value in result.items()}
-        for site in ['output_parameters', 'supercell', 'standardized_structure']:
-            result.pop(site, None)
+            result = get_marked_structures(input_structure, **inputs)
+            self.ctx.supercell = input_structure
+            self.ctx.equivalent_sites_data = result.pop('output_parameters').get_dict()
```

Comment:
Since `get_marked_structures()` doesn't return a `supercell_structure` or `symm_analysis_dict`, these can't be sent to output. This causes a problem, because these two are required outputs - thus the calculation using an `atoms_list` input returns error code 11 (missing required output(s)):

```
Property     Value
-----------  -------------------------------------------------------------
type         XpsWorkChain
state        Finished [11] The process did not register a required output.
pk           200
uuid         edb735da-d8f8-436f-b3bf-94519255f5f6
label        XpsWorkChain (default)
description
ctime        2023-06-27 10:06:51.852633+00:00
mtime        2023-06-27 10:18:55.460074+00:00

Inputs                    PK    Type
------------------------  ----  -------------
ch_scf
    clean_workdir         190   Bool
    max_iterations        189   Int
    kpoints_force_parity  188   Bool
    kpoints_distance      187   Float
    pw
        parallelization   186   Dict
        parameters        185   Dict
        structure         184   StructureData
        code              1     InstalledCode
        pseudos
            O             12    UpfData
            C             32    UpfData
            Li            30    UpfData
core_hole_pseudos
    O                     110   UpfData
    C                     106   UpfData
    Li                    107   UpfData
gipaw_pseudos
    O                     109   UpfData
    C                     105   UpfData
    Li                    108   UpfData
abs_atom_marker           191   Str
atoms_list                195   List
calc_binding_energy       192   Bool
clean_workdir             194   Bool
core_hole_treatments      196   Dict
correction_energies       193   Dict
spglib_settings           197   Dict
structure                 184   StructureData
voight_gamma              198   Float
voight_sigma              199   Float

Outputs                   PK    Type
------------------------  ----  ------
chemical_shifts
    Li_cls                250   Dict
    O_cls                 251   Dict
    C_cls                 252   Dict
final_spectra_cls
    Li_cls_spectra        253   XyData
    O_cls_spectra         254   XyData
    C_cls_spectra         255   XyData
output_parameters_ch_scf
    site_0                238   Dict
    site_4                242   Dict
    site_6                246   Dict

Called                   PK  Type
---------------------  ----  ----------------------
get_marked_structures   202  get_marked_structures
site_0_xps              208  PwBaseWorkChain
site_4_xps              210  PwBaseWorkChain
site_6_xps              212  PwBaseWorkChain
compile_final_spectra   249  get_spectra_by_element

```

Reply:
Either changing the outputs to not be required, or integrating `get_marked_structures()` into `get_xspectra_structures()` (which does return this information) should fix the issue.

Reply:
Thanks for pointing out this. I changed them to be not required.